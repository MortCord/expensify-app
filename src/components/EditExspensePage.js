import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { editExpense } from "../actions/expenses";
import { removeExpense } from "../actions/expenses";

export class EditExspensePage extends React.Component{
    onSubmit = (expense) =>{
        this.props.editExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    }

    onRemove = (expense) =>{
        this.props.removeExpense({id: this.props.expense.id});
        this.props.history.push('/');
    }

    render(){
        return(
            <div>
                <h1>Edit expense</h1>
                <ExpenseForm isAddOrEdit={"Edit"} onSubmit={this.onSubmit} expense={this.props.expense} />
                <button onClick={this.onRemove}>Remove</button>
        </div>
        )
    }
}

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});


const mapDispatchToProps = (dispatch, props) =>({
    editExpense: (id,expense) => dispatch(editExpense(id,expense)),
    removeExpense: (data) => dispatch(removeExpense(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExspensePage);