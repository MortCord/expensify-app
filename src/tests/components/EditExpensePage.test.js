import React from "react";
import { shallow } from "enzyme";
import { EditExspensePage } from "../../components/EditExspensePage";
import expenses from '../fixtures/expenses';

let editExpense, removeExpense, history, wrapper;

beforeEach(() =>{
    editExpense = jest.fn();
    removeExpense = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(<EditExspensePage editExpense={editExpense} removeExpense={removeExpense} history={history} expense={expenses[2]} />);
});

test('test render EditExpensePage', () =>{
    expect(wrapper).toMatchSnapshot();
});

test('test handle editExpense', ()=>{
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[2]);

    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(editExpense).toHaveBeenLastCalledWith(expenses[2].id, expenses[2]);
});

test('test handle removeExpense', ()=>{
    wrapper.find('button').simulate('click');

    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(removeExpense).toHaveBeenLastCalledWith({
        id: expenses[2].id
    });
});