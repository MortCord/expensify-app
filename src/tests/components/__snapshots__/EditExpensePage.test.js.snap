// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render EditExpensePage 1`] = `
<div>
  <h1>
    Edit expense
  </h1>
  <ExpenseForm
    expense={
      Object {
        "amount": 3,
        "createdAt": 345600000,
        "description": "desc3",
        "id": "3",
        "note": "",
      }
    }
    isAddOrEdit="Edit"
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
